subsys/usb/host/class/usbh_hub_mgr.h

```diff
@@ -0,0 +1,161 @@
+	HUB_PORT_RUN_PORT_ATTACHED,
+	HUB_PORT_RUN_CHECK_PORT_DETACH,
+	HUB_PORT_RUN_CHECK_CHILD_HUB,
+	HUB_PORT_RUN_GET_CONNECTION_BIT,
+	HUB_PORT_RUN_CHECK_CONNECTION_BIT,
+};
+
+enum usbh_hub_prime_status {
+	HUB_PRIME_NONE = 0,
+	HUB_PRIME_CONTROL,
+	HUB_PRIME_PORT_CONTROL, 
+	HUB_PRIME_INTERRUPT,
+};
+
+/* Port instance structure (unified structure with necessary fields) */
+struct usbh_hub_port_instance {
+	struct usb_device *udev;		/* Connected USB device */
+	uint8_t port_status;			/* Port application status */
+	enum usbh_port_state state;		/* Port overall state */
+	uint8_t reset_count;			/* Reset retry count */
+	uint8_t speed;				/* Device speed */
+	uint8_t port_num;			/* Port number */
+};
+
+/**
+ * @brief HUB management data structure
+ */
+struct usbh_hub_mgr_data {
+	struct usb_device *hub_udev;
+	struct usbh_context *uhs_ctx;
+
+	/* Hub instance data */
+	struct usbh_hub_instance hub_instance;
+
+	/* Hub state management */
+	enum usbh_hub_state state;
+	enum usbh_hub_app_status hub_status;
+
+	/* Port management - unified using port_list */
+	struct usbh_hub_port_instance *port_list;	/* Port instance list */
+	uint8_t num_ports;				/* Total number of ports */
+	uint8_t current_port;				/* Currently processing port */
+	uint8_t port_index;				/* Port index */
+
+	/* Work items for state machine processing */
+	struct k_work_delayable hub_work;
+	struct k_work_delayable port_work;
+
+	/** Interrupt endpoint descriptor */
+	struct usb_ep_descriptor *int_ep;
+	/** Device connection status */
+	bool connected;
+	/** Interrupt transfer active flag */
+	bool int_active;
+
+	/* Hierarchy structure management fields */
+	struct usbh_hub_mgr_data *parent_hub;
+	uint8_t parent_port;
+	sys_slist_t child_hubs;
+	sys_snode_t child_node;
+	uint8_t hub_level;
+
+	/* Global Hub chain management list node */
+	sys_snode_t node;
+
+	/* Synchronization and state management */
+	struct k_mutex lock;
+	bool being_removed;
+	uint32_t last_error_time;
+	uint8_t error_count;
+	uint8_t prime_status;
+
+	struct uhc_transfer *interrupt_transfer;
+	uint8_t int_buffer[8];
+};
+
+/* Hub manager API function declarations */
+
+/**
+ * @brief Find corresponding Hub management data by USB device
+ */
+struct usbh_hub_mgr_data *usbh_hub_mgr_get_hub_by_device(struct usb_device *udev);
+
+/**
+ * @brief Get high-speed Hub address (for split transactions)
+ */
+uint8_t usbh_hub_mgr_get_hs_hub_addr(struct usb_device *udev);
+
+/**
+ * @brief Get high-speed Hub port (for split transactions)
+ */
+uint8_t usbh_hub_mgr_get_hs_hub_port(struct usb_device *udev);
+
+/**
+ * @brief Print Hub topology structure (for debugging)
+ */
+void usbh_hub_mgr_print_topology(void);
+
+/**
+ * @brief Get Hub statistics information
+ */
+void usbh_hub_mgr_get_stats(struct usbh_hub_stats *stats);
+
+/**
+ * @brief Check if Hub chain depth exceeds limit
+ */
+static inline bool usbh_hub_mgr_check_depth_limit(uint8_t hub_level)
+{
+	return hub_level < CONFIG_USBH_HUB_MAX_LEVELS;
```

Comment:
Assuming `CONFIG_USBH_HUB_MAX_LEVELS=3`, if the situation is like this:

```
L0
|
L1
|
L2
```

Then this function works, but if the situation is like this:

```
L0
| \
L1 L1
```
Then adding a L2 HUB would not work, because there are not enough `USBH_DEFINE_HUB_CLASS()` called in the "LISTIFY" above.

The maximum number of HUBs in total is reached, but the maximum depth is not reached.

Is it really needed to check the max depth other than what the USB standard proposes?
It looks like it is only needed to check the max number of HUB in total?
It is possible that I miss the context though.